doc/build-helpers/images/dockertools.section.md

```diff
@@ -36,84 +48,216 @@ buildImage {
+  If `false`, the symlinks will be transformed into directories.
+  This behaves the same as `rsync -k` when `keepContentsDirlinks` is `false`, and the same as `rsync -K` when `keepContentsDirlinks` is `true`.
+
+    _Default value:_ `false`.
+
+`runAsRoot` _optional_
+
+: A bash script that will run as root inside a VM that contains the existing layers of the base image and the new generated layer (including the files from `copyToRoot`).
+  This can be seen as an equivalent of `RUN ...` in a `Dockerfile`.
+  A value of `null` means that this step in the image generation process will be skipped.
+
+  :::{.caution}
+  Using this attribute requires the `kvm` device to be available.
+  :::
+
+  :::{.note}
+  At the time the script in `runAsRoot` is run, the files specified directly in `copyToRoot` will be present in the VM, but their dependencies might not be there yet.
+  Copying their dependencies into the generated image is a step that happens after `runAsRoot` finishes running.
+  :::
+
+    _Default value:_ `null`.
+
+`extraCommands` _optional_
+
+: A bash script that will run before the layer created by `buildImage` is finalised.
+  This is similar to `runAsRoot`, but the script specified in `extraCommands` is **not** run as root, and does not involve creating a VM.
+  It is simply run as part of building the derivation that outputs the layer created by `buildImage`.
+
+    _Default value:_ `""`.
+
+`config` _optional_
+
+: Used to specify the configuration of the containers that will be started off the generated image.
+  Must be an attribute set, with each attribute as listed in the [Docker Image Specification v1.3.0](https://github.com/moby/moby/blob/46f7ab808b9504d735d600e259ca0723f76fb164/image/spec/spec.md#image-json-field-descriptions).
+
+    _Default value:_ `null`.
+
+`architecture` _optional_
+
+: Used to specify the image architecture.
+  This is useful for multi-architecture builds that don't need cross compiling.
+  If specified, its value should follow the [OCI Image Configuration Specification](https://github.com/opencontainers/image-spec/blob/main/config.md#properties), which should still be compatible with Docker.
+
+    _Default value:_ the same value from `pkgs.go.GOARCH`.
+
+`diskSize` _optional_
+
+: Controls the disk size (in megabytes) of the VM used to run the script specified in `runAsRoot`.
+  This attribute is ignored if `runAsRoot` is `null`.
+
+    _Default value:_ 1024.
+
+`buildVMMemorySize` _optional_
+
+: Controls the amount of memory (in megabytes) provisioend for the VM used to run the script specified in `runAsRoot`.
+  This attribute is ignored if `runAsRoot` is `null`.
+
+    _Default value:_ 512.
+
+`created` _optional_
+
+: Specifies the time of creation of the generated image.
+  This should be either an ISO-8601-formatted date and time or `"now"`, in which case `buildImage` will use the current date.
```

Comment:
If you can find a good link to the ISO8601 spec, would be good to add here

Reply:
```suggestion
  This should be either a date and time formatted according to [ISO-8601](https://en.wikipedia.org/wiki/ISO_8601)  or `"now"`, in which case `buildImage` will use the current date.
```